.github/workflows/test-upstream.yaml

```diff
@@ -140,7 +140,7 @@ jobs:
       # the cache will still speedup the rest of the installation
       - name: install linkml
         working-directory: linkml
-        run: poetry sync --no-interaction --all-extras
+        run: poetry install --no-interaction --all-extras
```

Comment:
Why did you switch to `poetry install`? 
`poetry sync` avoids the presence of [untracked outdated packages](https://python-poetry.org/docs/cli/#install) (here from cache).